From 93c6099369fa40187d59079ba75961972d96b33f Mon Sep 17 00:00:00 2001
From: Joseph Chen <chenjh@rock-chips.com>
Date: Mon, 28 May 2018 14:57:19 +0800
Subject: [PATCH] power: rk816-battery: fix CDP charger recoginze as DCP
 charger

CDP charger should be treat as SDP charger, but charge current 1.5A(not 450mA).

Change-Id: I9892e8ce7cb26a1990b344a9ea83ce55a9a1b81a
Signed-off-by: Joseph Chen <chenjh@rock-chips.com>
---
 drivers/power/rk816_battery.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/power/rk816_battery.c b/drivers/power/rk816_battery.c
index 4ef0585f1ca0..62aa6fd13c19 100644
--- a/drivers/power/rk816_battery.c
+++ b/drivers/power/rk816_battery.c
@@ -1442,8 +1442,15 @@ static void rk816_bat_set_chrg_param(struct rk816_battery *di,
 			rk816_bat_set_current(di, INPUT_CUR450MA);
 		power_supply_changed(di->usb);
 		break;
-	case USB_TYPE_AC_CHARGER:
 	case USB_TYPE_CDP_CHARGER:
+		di->usb_in = 1;
+		di->ac_in = 0;
+		di->prop_status = POWER_SUPPLY_STATUS_CHARGING;
+		if (di->dc_in == 0)
+			rk816_bat_set_current(di, INPUT_CUR1500MA);
+		power_supply_changed(di->usb);
+		break;
+	case USB_TYPE_AC_CHARGER:
 		di->ac_in = 1;
 		di->usb_in = 0;
 		di->prop_status = POWER_SUPPLY_STATUS_CHARGING;
-- 
2.35.3

